tweets.py

```diff
@@ -59,30 +67,85 @@ def get_observations(place):
 # however, the observations aren't necessarily spaced equally throughout
 # the day. do something more sophisticated later!
 def get_daily_temp(place):
-    observations = get_observations(place)
-
-    temps = [temp for (timestamp, temp) in observations]
+    temps = [temp for (timestamp, temp) in observations if temp]
     average = list_average(temps)
     average_fahrenheit = average * 1.8 + 32
 
     return average_fahrenheit
 
-# This currently calculates the average temperature for the current day
-def get_historical_temps(place):
-    historical_data_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
+def get_historical_temps(place, local_time):
+    place_id = place['USAF'] + "-" + place['WBAN']
+
+    current_day_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
         data_url=DATA_URL,
-        id=place['USAF'] + "-" + place['WBAN'],
+        id=place_id,
         month='{:02}'.format(END_TIME.month),
         day='{:02}'.format(END_TIME.day)
     )
-    response = requests.get(historical_data_url)
-    lines = (line.decode('utf-8') for line in response.iter_lines())
-    reader = csv.reader(lines)
-    next(reader, None)  # skip the headers
+
+    current_day = get_reader(current_day_url)
+
+    # we want the last 24 hours of weather data, ending at 6pm today local time
+    tz_offset = int(local_time.utcoffset().total_seconds() / 3600)
+    additional_hours_needed = -6 - tz_offset
+
+    if additional_hours_needed < 0:
+        # we need to grab the previous day's sheet as well and take the last
+        # additional_hours_needed hours for each year
+        previous_day = END_TIME - Timedelta(days=1)
+        previous_day_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
+            data_url=DATA_URL,
+            id=place_id,
+            month='{:02}'.format(previous_day.month),
+            day='{:02}'.format(previous_day.day)
+        )
+
+        previous_day_filtered = [
+            row for row
+            in get_reader(previous_day_url)
+            if int(row[1]) >= (24 + additional_hours_needed)
+        ]
+
+        current_day_filtered = [
+            row for row
+            in get_reader(previous_day_url)
+            if int(row[1]) < (24 + additional_hours_needed)
+        ]
```

Comment:
If you use pandas to read the csv (you can pass a URL to `pd.read_csv()`) you can do this filtering more easily:

```
current_day = pd.read_csv(current_day_url)
current_day[current_day.hour < END_TIME.hour]
```

Reply:
I like `requests` but if you feel strongly about this syntax you're welcome to rewrite it